app/src/main/java/app/musikus/ui/theme/Theme.kt

```diff
@@ -74,43 +79,142 @@ fun MusikusTheme(
-
-        val navBarColorDarkTheme = android.graphics.Color.TRANSPARENT
-        val navBarColorLightTheme = android.graphics.Color.TRANSPARENT
-        val navBarColorLightThemeOldDevices = DarkColorScheme.surface.toArgb()
-
-        context.enableEdgeToEdge(
-            statusBarStyle = if (useDarkTheme)
-                SystemBarStyle.dark(
-                    scrim = statusBarColorDarkTheme
-                )
+    // skip this block when in LocalInspectionMode (Compose Preview) so that we can use
+    // MusikusTheme{} wrapper also in Compose Previews
+    if (!LocalInspectionMode.current) {
+        val context = LocalContext.current as ComponentActivity
+        LaunchedEffect(key1 = theme) {
+
+            val statusBarColorDarkTheme = android.graphics.Color.TRANSPARENT
+            val statusBarColorLightTheme = android.graphics.Color.TRANSPARENT
+            val statusBarColorLightThemeOldDevices = DarkColorScheme.surface.toArgb()
+
+            val navBarColorDarkTheme = android.graphics.Color.TRANSPARENT
+            val navBarColorLightTheme = android.graphics.Color.TRANSPARENT
+            val navBarColorLightThemeOldDevices = DarkColorScheme.surface.toArgb()
+
+            context.enableEdgeToEdge(
+                statusBarStyle = if (useDarkTheme)
+                    SystemBarStyle.dark(
+                        scrim = statusBarColorDarkTheme
+                    )
                 else SystemBarStyle.light(
                     scrim = statusBarColorLightTheme,
                     darkScrim = statusBarColorLightThemeOldDevices
                 ),
-            navigationBarStyle = if (useDarkTheme)
-                SystemBarStyle.dark(
-                    scrim = navBarColorDarkTheme
-                )
+                navigationBarStyle = if (useDarkTheme)
+                    SystemBarStyle.dark(
+                        scrim = navBarColorDarkTheme
+                    )
                 else SystemBarStyle.light(
                     scrim = navBarColorLightTheme,
                     darkScrim = navBarColorLightThemeOldDevices
                 ),
-        )
+            )
+        }
     }
-
     MaterialTheme(
         colorScheme = colorSchemeDarkOrLight,
         content = content
     )
 }
 
+@Composable
+fun MusikusThemedPreview(
+    theme: ColorSchemeSelections = ColorSchemeSelections.DEFAULT,
+    content: @Composable () -> Unit,
+) {
+    MusikusTheme(
+        theme = ThemeSelections.SYSTEM,
+        colorScheme = theme
+    ) {
+        Surface {
+            content()
+        }
+    }
+}
+
+class MusikusColorSchemeProvider : PreviewParameterProvider<ColorSchemeSelections> {
+    override val values = sequenceOf(
+        ColorSchemeSelections.MUSIKUS,
+        ColorSchemeSelections.DYNAMIC,
+        ColorSchemeSelections.LEGACY,
+    )
+}
+
+/** Convenience Templates for predefined Compose Preview groups **/
+
+private const val groupNameAllLight: String = "All Elements - Light"
```

Comment:
This whole block seems pretty hacky to me... Why do we need all these different annotation classes `MusikusPreviewElementX`?
And don't you think, we could store these in a separate file? 😅

Reply:
Hm you're right it's a bit "Hacky" but it's the best solution there is now for using Android Studio Preview Grouping. These values are used in the `@Preview` Annotations and have to be "hardcoded", it's not possible to use a parameter...

The purpose is to have "a bunch of Slots" for different UI elements to have previews. These Group arguments are used to group different styles of the same element together in Android Studio (dark style + color schemes + eventually different screen sizes):

![image](https://github.com/matthiasemde/musikus-android/assets/28269133/c611dc8e-c70b-4abd-a6aa-a5b67c4c2fc7)


for example, "Element 5" is the Library Item. The `group` argument makes Android Studio group them all together so they can be displayed together: 

![image](https://github.com/matthiasemde/musikus-android/assets/28269133/8a270def-cc55-45cb-958c-cf333a5590cd|width=100px)


This is "the best I found" - with the only downside of having to remember which element is "Element X" in each screen - but you can quickly skip through. If you have a better solution for this problem, I'd be glad for ideas :)

Reply:
> And don't you think, we could store these in a separate file? 😅

I now created an extra file, `PreviewStyling.kt`